### PR TITLE
Don't use fix1 for enzyme

### PR DESCRIPTION
**Pull request by wsmoses**

### CODE REVIEW

**Comment by devmotion** [comment]
Thanks, that's much better 🙂 Does this also fix an open issue?  Maybe even the Turing-related one?

**Comment by wsmoses** [comment]
Not yet, am trying to minimize that presently.

File: src/AD_Enzyme.jl [hunk 1]
```diff
@@ -43,7 +43,8 @@ function logdensity_and_gradient(∇ℓ::EnzymeGradientLogDensity{<:Any,<:Enzyme
                                  x::AbstractVector)
     @unpack ℓ, mode, shadow = ∇ℓ
     _shadow = shadow === nothing ? Enzyme.onehot(x) : shadow
-    y, ∂ℓ_∂x = Enzyme.autodiff(mode, Base.Fix1(logdensity, ℓ), Enzyme.BatchDuplicated,
+    y, ∂ℓ_∂x = Enzyme.autodiff(mode, logdensity, Enzyme.BatchDuplicated,
+                               Const(ℓ),
```
Comment (devmotion) [inline src/AD_Enzyme.jl:47]:
```suggestion
                               Enzyme.Const(ℓ),
```

File: src/AD_Enzyme.jl [hunk 2]
```diff
@@ -55,7 +56,8 @@ function logdensity_and_gradient(∇ℓ::EnzymeGradientLogDensity{<:Any,<:Enzyme
     # Ref: https://github.com/EnzymeAD/Enzyme.jl/issues/107
     y = logdensity(ℓ, x)
     ∂ℓ_∂x = zero(x)
-    Enzyme.autodiff(mode, Base.Fix1(logdensity, ℓ), Enzyme.Active,
+    Enzyme.autodiff(mode, logdensity, Enzyme.Active,
+                    Const(ℓ),
```
Comment (devmotion) [inline src/AD_Enzyme.jl:60]:
```suggestion
                    Enzyme.Const(ℓ),
```

**Comment by devmotion** [comment]
> Not yet, am trying to minimize that presently.

BTW I saw your comment in the issue in the Enzyme repo but unfortunately I haven't been able to debug and minimize it further yet, I only have very limited access to my computer during the Christmas break.

**Review by devmotion — APPROVED** [review]

**Pull request merged**